integration/v3_stm_test.go

```diff
@@ -197,3 +197,50 @@ func TestSTMSerialize(t *testing.T) {
 		}
 	}
 }
+
+// TestSTMApplyOnConcurrentDeletion ensures that concurrent key deletion
+// failed the GET operation within STM and triggers retry.
+func TestSTMApplyOnConcurrentDeletion(t *testing.T) {
+	clus := NewClusterV3(t, &ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+
+	etcdc := clus.RandClient()
+	if _, err := etcdc.Put(context.Background(), "foo", "bar"); err != nil {
```

Comment:
might want to TODO() these so the ctxs are consistent across the test, but no strong opinion